src/gluonts/model/forecast_generator.py

```diff
@@ -145,3 +183,46 @@ def __call__(
                     info=batch["info"][i] if "info" in batch else None,
                 )
             assert i + 1 == len(batch["forecast_start"])
+
+
+class DistributionForecastGenerator(ForecastGenerator):
+    @validated()
+    def __init__(self, distr_output) -> None:
+        self.distr_output = distr_output
+
+    def __call__(
+        self,
+        inference_data_loader: DataLoader,
+        prediction_net,
+        input_names: List[str],
+        freq: str,
+        output_transform: Optional[OutputTransform],
+        num_samples: Optional[int],
```

Comment:
Among these arguments, it seems to me like only `inference_data_loader` is really required "at run time" -- everything else could maybe be fixed at construction time. But then the `Predictor` classes that use this (see eg [here](https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/mx/model/predictor.py#L150-L174)) are really left with very little purpose.